packages/examples/fortune-v2/job-launcher/server/src/common/decorators/user.ts

```diff
@@ -0,0 +1,15 @@
+import { createParamDecorator, ExecutionContext } from "@nestjs/common";
+import { UserStatus, UserType } from "../enums/user";
+import { IBase } from "./base";
+
+export interface IUser extends IBase {
+  password: string;
+  email: string;
+  status: UserStatus;
+  type: UserType;
+}
+
+export const User = createParamDecorator((_data: unknown, ctx: ExecutionContext) => {
+  const request = ctx.switchToHttp().getRequest();
```

Comment:
maybe also handling non Http contexts, like GraphQL or Websockets

Reply:
Lets leave it as improvements, I will add it to backlog